processor/src/lib.rs

```diff
@@ -253,6 +253,18 @@ where
             return Err(ExecutionError::ProgramAlreadyExecuted);
         }
 
+        // Load the program's advice data into the advice provider
+        for (digest, values) in program.mast_forest().advice_map().clone().into_iter() {
```

Comment:
similar here, we can avoid the `clone()` by implementing `iter()`

Reply:
Done.